src/rprblender/nodes/blender_nodes.py

```diff
@@ -972,42 +973,54 @@ def export_hybrid(self):
         return shader2
 
 
-class ShaderNodeTexCoord(RuleNodeParser):
+class ShaderNodeTexCoord(NodeParser):
     # outputs: Generated, Normal, UV, Object, Camera, Window, Reflection
     # Supported outputs by RPR: Normal, UV
 
-    nodes = {
-        "Generated": {
-            "type": pyrpr.MATERIAL_NODE_INPUT_LOOKUP,
-            "params": {
-                pyrpr.MATERIAL_INPUT_VALUE: pyrpr.MATERIAL_NODE_LOOKUP_UV,
-            },
-            "warn": "TexCoord Generated output is not supported, UV will be used"
-        },
-        "Normal": {
-            "type": pyrpr.MATERIAL_NODE_INPUT_LOOKUP,
-            "params": {
+    def export(self):
+        tex_coord_type = self.socket_out.name
+
+        if tex_coord_type == 'Generated':
+            data = self.create_node(pyrpr.MATERIAL_NODE_INPUT_LOOKUP, {
+                pyrpr.MATERIAL_INPUT_VALUE: pyrpr.MATERIAL_NODE_LOOKUP_P_LOCAL,
+            })
+            if self.object:
+                # normalize over object bounding box
+                # get min and max of boinding box (is there a faster way?)
+                min = [sys.maxsize] * 3
+                max = [-sys.maxsize] * 3
+                for point in self.object.bound_box:
```

Comment:
Is there a faster way to do this in python?

Reply:
yes:
mn = tuple(min(p[i] for p in self.object.bound_box) for i in range(3))
mx = tuple(max(p[i] for p in self.object.bound_box) for i in range(3))

Reply:
thanks.